src/assets/data-forms.js

```diff
@@ -0,0 +1,155 @@
+// application state object
+const appState = {
+  percentageInfected: 100,
+  stateCode: "",
+  incarceratedPopulation: 0,
+  incarceratedPopulationMax: 0,
+  incarceratedPopulationMin: 0
+};
+
+const stateNames = Object.values(ICU_DATA).map(function(record) {
+  return record.name;
+});
+
+const repaintFunctions = [];
+
+function registerRepaintFunction(fn) {
+  // these functions will all receive the state code as first argument,
+  // so they should either use it or ignore it gracefully
+  repaintFunctions.push(fn);
+}
+
+function repaint() {
+  const stateCode = appState.stateCode;
+  repaintFunctions.forEach(function(fn) {
+    fn(stateCode);
+  });
+}
+
+const stateCodesByName = {};
+Object.entries(ICU_DATA).forEach(function(entry) {
+  const code = entry[0];
+  const name = entry[1].name;
+  stateCodesByName[name] = code;
+});
+
+function updateAppState(changesObj) {
+  Object.assign(appState, changesObj);
+  repaint();
+}
+
+function getStateName(stateCode) {
+  return ICU_DATA[stateCode].name;
+}
+
+function getStateCodeFromName(name) {
+  return stateCodesByName[name];
+}
+
+function getIncarceratedPopulation(stateCode) {
+  return ICU_DATA[stateCode].incarceratedPopulation;
+}
+
+function getNumberOfICUBeds(stateCode) {
+  return ICU_DATA[stateCode].numberOfICUBeds;
+}
+
+function getPercentageHospitalized(stateCode) {
+  // Returning a hard coded value for now, but you could replace this value with
+  // something that is populated from a form element, such as a slider.
+  return 0.05;
+}
+
+function getPercentageInfectedAsDecimal() {
+  return appState.percentageInfected / 100;
+}
+
+function getICUBedsPercentage(stateCode) {
+  var incarceratedPopulation = appState.incarceratedPopulation;
+  var numberOfICUBeds = getNumberOfICUBeds(stateCode);
+  var percentageHospitalized = getPercentageHospitalized(stateCode);
+  var percentageInfected = getPercentageInfectedAsDecimal();
+
+  return parseInt(
+    ((incarceratedPopulation * percentageInfected * percentageHospitalized) /
+      numberOfICUBeds) *
+      100
+  );
+}
+
+function paintHeading(stateCode) {
+  var headingText;
+
+  if (stateCode == "US") {
+    headingText =
+      "As COVID-19 spreads, prisons and jails are the last dense gatherings in America. " +
+      "If states don't act now, " +
+      getICUBedsPercentage(stateCode) +
+      "% of ICU beds nationwide will be needed just for the infirm from prisons and jails.";
+  } else {
+    headingText =
+      "If no action is taken, " +
+      getICUBedsPercentage(stateCode) +
+      "% of " +
+      getStateName(stateCode) +
+      "’s ICU beds will be needed just for the infirm from prisons and jails.";
```

Comment:
beware that I was once a professional copy editor and cannot be out-nitpicked on matters of style and grammar 😈  .... do we have a house style that specifies this? because depending on that, `s's` might be the preferred form (e.g. in Chicago style)

Reply:
You got me. You outduelled me. We don't have a house style, and I see now that there is pretty solid consensus around sticking with `s's` in this case. Don't mind me!